ReCaptchaFrontendUi/view/frontend/web/js/nonInlineReCaptchaRenderer.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+define([
+    'jquery'
+], function ($) {
+    'use strict';
+
+    var reCaptchaEntities = [],
+        initialized = false,
+        rendererRecaptchaId = 'recaptcha-invisible',
+        rendererReCaptcha = null;
+
+    return {
+        add: function (reCaptchaEntity, parameters) {
+            if (parameters.size === 'invisible' && parameters.badge !== 'inline') {
+                if (!initialized) {
+                    this._init();
+                    grecaptcha.render(rendererRecaptchaId, parameters);
+                    setInterval(this._resolveVisibility, 100);
```

Comment:
Why do we need `setInterval`? Using timers in general bad practice, moreover, in this case, we never a clean interval and we will call `_resolveVisibility` infinite time. 
As I see, grecaptcha has callback function that called when it will be rendered, we should use it, and start to execute out login only after grecaptcha will be rendered.